internal/provider/kubernetes/controller.go

```diff
@@ -45,17 +45,18 @@ import (
 )
 
 type gatewayAPIReconciler struct {
-	client          client.Client
-	log             logging.Logger
-	statusUpdater   Updater
-	classController gwapiv1.GatewayController
-	store           *kubernetesProviderStore
-	namespace       string
-	namespaceLabel  *metav1.LabelSelector
-	envoyGateway    *egv1a1.EnvoyGateway
-	mergeGateways   sets.Set[string]
-	resources       *message.ProviderResources
-	extGVKs         []schema.GroupVersionKind
+	client            client.Client
+	log               logging.Logger
+	statusUpdater     Updater
+	classController   gwapiv1.GatewayController
+	store             *kubernetesProviderStore
+	namespace         string
+	namespaceLabel    *metav1.LabelSelector
+	envoyGateway      *egv1a1.EnvoyGateway
+	mergeGateways     sets.Set[string]
+	resources         *message.ProviderResources
+	extGVKs           []schema.GroupVersionKind
+	extServerPolicies []schema.GroupVersionKind
```

Comment:
why not reuse `extGVKs` here ?

Reply:
We should keep them separate since they are for different things and the extGVKs go to the post route modify hook, and extServerPolicies go to the post listener hook. Makes it easier to check which is which.

Reply:
Like @AliceProxy wrote, keeping it separate captures the fact that these GVKs are used for different purposes in different places.

Maybe I should rename `extGVKs` to `extRouteFilters` ?